src/optlang/matrix_interface.py

```diff
@@ -0,0 +1,964 @@
+            The new name of the constraint.
+
+        Returns
+        -------
+        Nothing.
+
+        """
+        self.reset()
+        self.constraints.remove(old)
+        self.constraints.add(new)
+        name_map = {k: k for k in self.constraints}
+        name_map[old] = new
+        self.constraint_coefs = {
+            (name_map[k[0]], k[1]): v for k, v in self.constraint_coefs.items()
+        }
+        self.constraint_lbs[new] = self.constraint_lbs.pop(old)
+        self.constraint_ubs[new] = self.constraint_ubs.pop(old)
+
+    def rename_variable(self, old, new):
+        """Rename a variable.
+
+        Parameters
+        ----------
+        old : str
+            The old name of the variable.
+        new : str
+            The new name of the variable.
+
+        Returns
+        -------
+        Nothing.
+
+        """
+        self.reset()
+        self.variables.remove(old)
+        self.variables.add(new)
+        name_map = {k: k for k in self.variables}
+        name_map[old] = new
+        self.constraint_coefs = {
+            (k[0], name_map[k[1]]): v for k, v in self.constraint_coefs.items()
+        }
+        self.obj_quadratic_coefs = {
+            (name_map[k[0]], name_map[k[1]]): v
+            for k, v in self.obj_quadratic_coefs.items()
+        }
+        self.obj_linear_coefs = {
+            name_map[k]: v for k, v in self.obj_linear_coefs.items()
+        }
+        self.variable_lbs[new] = self.variable_lbs.pop(old)
+        self.variable_ubs[new] = self.variable_ubs.pop(old)
+
+
+class Variable(interface.Variable):
+    def __init__(self, name, *args, **kwargs):
+        super(Variable, self).__init__(name=name, **kwargs)
+
+    @interface.Variable.type.setter
+    def type(self, value):
+        if self.problem is not None:
+            if value not in _TYPES:
+                raise ValueError(
+                    "This interface cannot handle variables of type '%s'. " % value
+                    + "The following variable types are available: "
+                    + ", ".join(_TYPES)
+                )
+        super(Variable, Variable).type.fset(self, value)
+
+    def _get_primal(self):
+        return self.problem.problem.primals.get(self.name, None)
+
+    @property
+    def dual(self):
+        if self.problem is not None:
+            return self.problem.problem.vduals.get(self.name, None)
+        return None
+
+    @interface.Variable.name.setter
+    def name(self, value):
+        old_name = getattr(self, "name", None)
+        super(Variable, Variable).name.fset(self, value)
+        if getattr(self, "problem", None) is not None:
+            if old_name != value:
+                self.problem.problem.rename_variable(old_name, value)
+
+
+class Constraint(interface.Constraint):
+    _INDICATOR_CONSTRAINT_SUPPORT = False
+
+    def __init__(self, expression, sloppy=False, *args, **kwargs):
+        super(Constraint, self).__init__(expression, *args, sloppy=sloppy, **kwargs)
```

Comment:
It's safer to not allow `*args` in class init, if possible.

```suggestion
    def __init__(self, expression, sloppy=False, **kwargs):
        super().__init__(expression=expression,, sloppy=sloppy, **kwargs)
```